src/services/getEditsForFileRename.ts

```diff
@@ -31,23 +31,20 @@ namespace ts {
     }
 
     function getImportsToUpdate(program: Program, oldFilePath: string, host: LanguageServiceHost): ReadonlyArray<ToUpdate> {
-        const checker = program.getTypeChecker();
         const result: ToUpdate[] = [];
         for (const sourceFile of program.getSourceFiles()) {
             for (const ref of sourceFile.referencedFiles) {
-                if (!program.getSourceFileFromReference(sourceFile, ref) && resolveTripleslashReference(ref.fileName, sourceFile.fileName) === oldFilePath) {
+                if (resolveTripleslashReference(ref.fileName, sourceFile.fileName) === oldFilePath) {
                     result.push({ sourceFile, toUpdate: ref });
                 }
             }
 
             for (const importStringLiteral of sourceFile.imports) {
-                // If it resolved to something already, ignore.
-                if (checker.getSymbolAtLocation(importStringLiteral)) continue;
-
                 const resolved = host.resolveModuleNames
                     ? host.getResolvedModuleWithFailedLookupLocationsFromCache && host.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName)
                     : program.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName);
-                if (resolved && contains(resolved.failedLookupLocations, oldFilePath)) {
+                // We may or may not have picked up on the file being renamed, so maybe successfully resolved to oldFilePath, or maybe that's in failedLookupLocations
+                if (resolved && contains([resolved.resolvedModule && resolved.resolvedModule.resolvedFileName, ...resolved.failedLookupLocations], oldFilePath)) {

```

Comment:
I would have you rather check resolvedModule.resolvedFileName separately and failedLookups separately to ensure not creating duplicate array just to check if it has the value.